cilium/cmd/bpf_ipcache_get.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"strings"
+
+	"github.com/cilium/cilium/common"
+	"github.com/cilium/cilium/pkg/maps/ipcache"
+
+	"github.com/hashicorp/go-immutable-radix"
+	"github.com/spf13/cobra"
+)
+
+const usage = "IP address must be in dotted decimal (192.168.1.1) or IPv6 (feab::f02b) form"
+
+var bpfIPCacheGetCmd = &cobra.Command{
+	Use:   "get",
+	Short: "Retrieve identity for an ip",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium bpf ipcache get")
+
+		if len(args) < 1 || args[0] == "" {
+			Usagef(cmd, "No ip provided. "+usage)
+		}
+
+		arg := args[0]
+
+		ip := net.ParseIP(arg)
+		if ip == nil {
+			Usagef(cmd, "Invalid ip address. "+usage)
+		}
+
+		bpfIPCacheList := dumpIPCache()
+
+		if len(bpfIPCacheList) == 0 {
+			fmt.Fprintf(os.Stderr, "No entries found.\n")
+			os.Exit(1)
+		}
+
+		value, exists := getLPMValue(ip, bpfIPCacheList)
+
+		if !exists {
+			fmt.Printf("%s does not map to any identity\n", arg)
+			os.Exit(1)
+		}
+
+		v := value.([]string)
+		if len(v) == 0 {
+			fmt.Printf("Unable to retrieve identity for LPM entry %s\n", arg)
+			os.Exit(1)
+		}
+
+		ids := strings.Join(v, ",")
+		fmt.Printf("%s maps to identity %s\n", arg, ids)
+	},
+}
+
+func init() {
+	bpfIPCacheCmd.AddCommand(bpfIPCacheGetCmd)
+}
+
+func dumpIPCache() map[string][]string {
+	bpfIPCacheList := make(map[string][]string)
+
+	if err := ipcache.IPCache.Dump(bpfIPCacheList); err != nil {
+		Fatalf("unable to dump IPCache: %s\n", err)
+	}
+
+	return bpfIPCacheList
```

Comment:
As a matter of style, it's good to avoid naming a map as a list as they're different underlying structure types. (Same for everywhere that is passing this map around and naming it `bpfIPCacheList`.)

Reply:
Fixed :)